protocol/messenger_communities.go

```diff
@@ -2761,7 +2918,59 @@ func (m *Messenger) HandleCommunityEventsMessageRejected(state *ReceivedMessageS
 		return nil
 	}
 
-	err = m.publishCommunityEvents(reapplyEventsMessage)
+	community, err := m.communitiesManager.GetByID(reapplyEventsMessage.CommunityID)
+	if err != nil {
+		return err
+	}
+
+	err = m.publishCommunityEvents(community, reapplyEventsMessage)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// HandleCommunityShardKey handles the private keys for the community shards
+func (m *Messenger) HandleCommunityShardKey(state *ReceivedMessageState, message *protobuf.CommunityShardKey, statusMessage *v1protocol.StatusMessage) error {
+	community, err := m.communitiesManager.GetByID(message.CommunityId)
```

Comment:
getbyid returns nil if the community is not in the db, so we need to handle it, how, we need to think about :)